modules/ROOT/pages/authorization-grant-types.adoc

```diff
@@ -0,0 +1,103 @@
+
+. Client app uses a login page supplied by the service provider to authenticate itself to the service provider.
+
+. After confirming that the client ID is valid (that is, the client app had previously registered with the service provider), and authenticating the resource owner's login credentials, the service provider returns an authorization code.
+
+. Client app calls the authorization server again, and provides its authorization code, client ID (again), and client secret.
+
+. Service provider returns an access token.
+
+. Client app calls the resource server, providing the access token, to request the protected resource.
+
+. Service provider delivers the protected resource.
+
+
+== Implicit
+
+As with the Authorization Code grant type, an Implicit grant type involves both an authorization server and a resource server. However, rather than taking the extra steps to authenticate a client application by using an authorization code and client secret, the service provider does the following:
+
+* Authenticates a client based upon its client ID.
+
+* Provides the access token to access the protected resource.
+
+The Implicit grant type optimizes performance for browser clients that use a scripting language, such as Javascript. However, this grant type can expose the access token to the resource owner or the resource owner's user-agent.
+
+An OAuth dance that uses an Implicit grant type works as follows:
+
+. Client app calls the service provider's authorization server, providing its client ID to confirm that it is a registered client of the service provider.
+
+. Client app uses a login page supplied by the service provider to ask that the resource owner authenticate itself to the service provider.
+
+. After confirming that the client ID is valid (that is, the client app had previously registered with the service provider), and authenticating the resource owner's login credentials, the service provider returns an access token.
+
+. Client app calls the resource server, providing the access token, to request the protected resource.
+
+. Service provider delivers the protected resource.
+
+
+== Resource Owner Password Credentials
+
+With the Resource Owner Password Credentials grant type, a client application demands that the resource owner share their service provider login credentials. The client app then uses these credentials to access the service provider and log into the resource owner's account. The service provider accepts the valid login credentials and grants the client app full access to the resource owner's account.
+
+This type of authorization optimizes performance at the expense of sharing secure access credentials. The resource owner must trust the client app not to share access to private data.
+
+An OAuth dance uses the Resource Owner Password Credentials grant type works as follows:
```

Comment:
"works as follows" -> "as follows" 

or maybe "An OAuth dance uses" -> "And OAuth dance that uses"

Reply:
OK

Reply:
OK

Reply:
OK